includes/class-cli-commands.php

```diff
@@ -134,7 +140,12 @@ public function insert_redirect( $args, $assoc_args ) {
 			$to = esc_url_raw( $args[1] );
 		}
 
-		$redirect = Utilities\insert_redirect( $from, $to, $code );
+		$preserve_parameters = filter_var( $assoc_args['preserve-parameters'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );
+		if ( $preserve_parameters === null ){
+			WP_CLI::error( sprintf( 'In order to preserve the parameters in a URL, the value must be a boolean, %s is invalid', $assoc_args['preserve-parameters'] ) );
```

Comment:
You have the parameter documented in the long description as optional, e.g. in square brackets, however this is going to show any time the option isn't set, or isn't set a valid value meaning it's not really optional.

With the suggestions I made above you can just read the value of `$assoc_args['preserve-parameters']` and you'll get a boolean, true if the parameter is present. This validation isn't necessary.

Reply:
`$preserve_parameters = filter_var( $assoc_args['preserve-parameters'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );` Here if a user had a value in place that isn't a boolean it would then default to false. So I added `FILTER_NULL_ON_FAILURE` for when a user have a non boolean value in place so it can be caught and referenced. I'm happy for it to fall back to `false` and remove this validation.

Reply:
With the updated long description documentation for the argument I'm pretty sure it'll never be able to be set to another value, WP CLI will take care of that. Try it and see what happens when you try passing different values, or no value, and omitting the flag altogether.